fastapi/dependencies/utils.py

```diff
@@ -276,7 +277,7 @@ def get_dependant(
         param_field = get_param_field(
             param=param, default_field_info=params.Query, param_name=param_name
         )
-        if param_name in path_param_names:
+        if param_name in path_param_names and not issubclass(param.annotation, JsonWrapper):
```

Comment:
I defer to @tiangolo, but I personally don't think we should add special handling for `JsonWrapper` here or below.

I don't believe this is a performance sensitive part of the codebase, but adding special handling for this type:

1. Adds an on-going maintenance burden where we have to have tests for `JsonWrapper` and non-`JsonWrapper` code paths
2. Adds an on-going maintenance burden where we need to handle this edge case any time we may want to refactor this code in the future
3. Opens the door for requests for special handling of other types in the future

Moreover, because pydantic now runs its tests against FastAPI, it could potentially make it more difficult for pydantic to refactor the `Json` class in the future.

I could be convinced otherwise if this was a common request, but so far that doesn't seem to be the case.

There are potentially alternative approaches to make this work though; I'll comment on them elsewhere.